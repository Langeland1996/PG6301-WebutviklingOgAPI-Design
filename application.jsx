import React from "react";

export function ListMovies() {
    return (
        <h1>
            Movies
        </h1>
    )
}