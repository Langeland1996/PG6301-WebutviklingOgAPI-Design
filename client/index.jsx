import React from "react";
import ReactDOM from "react-dom";

ReactDOM.render(<h1>Homepage</h1>, document.getElementById("app"));