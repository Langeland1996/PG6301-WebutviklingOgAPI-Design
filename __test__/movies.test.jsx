import React from "react";
import {createRoot} from "react-dom/client";
import {ListMovies, NewMovieForm} from "../application";
import {act} from "@testing-library/react";
import {Simulate} from "react-dom/test-utils";
import * as reactDOM from "react-dom";

describe("movie application", () => {

    it("check if body == movies", () => {
        const element = document.createElement("div");

        act(() => {
            createRoot(element).render(<ListMovies/>)
        })

        expect(element.querySelector("h1").innerHTML).toEqual("List movies")

    })



    it("check if snapshot is matching", () => {
        const element = document.createElement("div");

        act(() => {
            createRoot(element).render(<ListMovies/>)
        })

        expect(element.innerHTML).toMatchSnapshot();
    })

    it("show me movie form", () => {
        const element = document.createElement("div")

        act(() => {
            createRoot(element).render(<NewMovieForm/>)
        })

        expect(element.innerHTML).toMatchSnapshot();

    })

    it("submit new movie", async () => {
        const onAddMovie = jest.fn();
        const element = document.createElement("div")

        await act(() => {
            createRoot(element).render(<NewMovieForm onAddMovie={onAddMovie}/>)
        })

        await act(() => {
            Simulate.change(
                // finn elementet med date-testid=title og send en input event til det
                element.querySelector("[data-testid=title]"),
                {target: {value: "Movie 1"}}
            );
            Simulate.change(
                // finn elementet med date-testid=year og send en input event til det
                element.querySelector("[data-testid=year]"),
                {target: {value: "2022"}}
            );
        })

        // finn elementet av type form og send en submit event til det
        await act(() => {
            Simulate.submit(element.querySelector("form"));
        })

        expect(onAddMovie).toHaveBeenCalledWith({
            title: "Movie 1",
            year: "2022",
            plot: "",
        })
    })
})