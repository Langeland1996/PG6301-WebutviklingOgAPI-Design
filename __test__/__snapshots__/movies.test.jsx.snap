// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`movie application check if body == movies 1`] = `"<h1>Movies</h1>"`;

exports[`movie application check if snapshot is matching 1`] = `"<div><h1>List movies</h1><ul><li>Movie 1</li><li>Movie 1</li></ul></div>"`;

exports[`movie application show me movie form 1`] = `"<form><h1>Add new movie</h1><div>title: <input data-testid="title" value="title"></div><div>year: <input data-testid="year"></div><div>plot: <div><textarea data-testid="plot"></textarea></div></div></form>"`;
